### ISSUE TITLE
Predicting multiple response correlations

### ISSUE THREAD

**Issue by mvuorre** [paul-buerkner/brms#254]
Currently, brms syntax allows predicting the residual SD `sigma`:

```r
bf(y ~ x + (x|id), sigma ~ x)
```

And multivariate response models:

```r
bf(cbind(y1, y2) ~ x + (x|id))
```

But multiple response models can't include predictors for sigma, nor is it possible to model the response correlation on x (`rescor(y1, y2)`).

Something like this would be easy to use (if it's easy / possible to implement, that's another issue!):

```r
bf(cbind(y1, y2) ~ x + (x|id), sigma ~ x)
```
where `sigma` could include residuals for y1, y2, and the response correlation `rescor(y1, y2)`. Alternatively, the syntax could require / allow users to specify predictors on these three independently (perhaps that isn't too important?).

Thoughts?

**Comment by paul-buerkner** [comment]
I agree that this is desireble and it will certainly be possible once multivariate models are generally implemented. 

Implementing this separately from #3 would likely require too much special case coding, which I try to avoid in order to keep the package maintainable.

**Comment by paul-buerkner** [comment]
In the new dev version, `sigma` can now be predicted in multivariate models as well. However, this is not (yet) possible with `rescor`. The reason is that `rescor` is a correlation matrix with more than one unique entry in models with more than two response variables. It remains unclear to me, how to predict a correlation matrix or parts of it. Any suggestions are welcome!

**Comment by paul-buerkner** [comment]
`vignette("brms_distreg")` explains the syntax. See also https://vuorre.netlify.com/post/2017/how-to-compare-two-groups-with-robust-bayesian-estimation-using-r-stan-and-brms/

**Comment by suldauna** [comment]
So to confirm, code for a bivariate response model in which residual variances are expected to vary by gender would look something like the following:

```
	bf_y1 <- bf (y1 ~ gender) + lf(sigma ~ gender)
	bf_y2 <- bf (y2 ~ gender) + lf(sigma ~ gender)
	m1 <- brm ( bf_y1 + bf_y2, data = dat, chains = 2, cores = 2)
```
Is that right?

And then all reported sigmas are on the log scale, correct?

**Comment by paul-buerkner** [comment]
looks correct to me.

suldauna <notifications@github.com> schrieb am Di., 17. Apr. 2018, 20:51:

**Comment by suldauna** [comment]
Regarding the above question about predicting a correlation matrix, perhaps that would be less problematic in a bivariate probit model in which the respective sigmas are fixed at 1 and only the covariance is allowed to vary freely.

**Comment by paul-buerkner** [comment]
It would be easier in any bivariate model, where we only have one correlation. Still, I am looking for a general solution in brms and I have yet to find a good way of doing it.

**Comment by stephensrmmartin** [comment]
@paul-buerkner This isn't for "predicting" correlations in a matrix, per se, but if partial pooling of a correlation matrix is desired across groups - Ben goodrich suggested a method to me that worked fairly well. See thread here: http://discourse.mc-stan.org/t/hierarchical-prior-for-partial-pooling-on-correlation-matrices/4852

**Comment by paul-buerkner** [comment]
Thanks Stephen, I will take a look!

**Comment by jackobailey** [comment]
@paul-buerkner, this is just an idle thought, but could you model the correlation matrix by assuming that it comes from an LKJ distribution (like we use as a prior anyway) and then modelling the LKJ distribution's scalar parameter using a log link (to keep it positive)?

**Comment by paul-buerkner** [comment]
Yes, but that would prevent us from having a separate model for each correlation and it would only control the scale not the location of the prior.

**Comment by jackobailey** [comment]
I might be missing something, but in that case why not simply create a vector of correlation parameters the same length as the number of comparisons between response variables and use them to fill in the correlation matrix? Then you could let users model, say, `cor_a_b` or `cor_b_c`, etc.

**Comment by paul-buerkner** [comment]
Then we don't ensure that the created correlation matrix is actually postive definite, a prerequisite for it being an actually valid correlation matrix.

**Comment by jeremy-koster** [comment]
Some co-authors and I currently are working on implementing unconventional correlation matrices that fulfill the positive definite requirement, but it's a special case written in Stan that I'm not sure would generalize easily to the brm() function.

**Comment by paul-buerkner** [comment]
Great! Once you have something you can share, I am happy to take a look and see if we can generalize it to the needs of brms (or other regression frameworks).

**Comment by bwiernik** [comment]
Here is an approach to modeling correlation matrices that comes from the meta-analysis literature https://onlinelibrary.wiley.com/doi/full/10.1002/jrsm.1214

**Comment by shannonajw** [comment]
@paul-buerkner This might be useful for a way to model correlation matrices: https://gitlab.com/dschrage/rcovreg
